src/components/button/button.scss

```diff
@@ -187,34 +187,82 @@ button.md-button.md-fab {
   }
 }
 
-.md-toast-open-top {
-  .md-button.md-fab-top-left,
-  .md-button.md-fab-top-right {
-    transition: $swift-ease-out;
-    transform: translate3d(0, $button-fab-toast-offset, 0);
-    &:not([disabled]) {
-      &.md-focused,
-      &:hover {
-        transform: translate3d(0, $button-fab-toast-offset - 1, 0);
-      }
+/*
+ * When a toast is open that is on the same side as our FAB, move the FAB out of the way.
+ *
+ * Note: On mobile; the toast will always open with the `md-toast-open-bottom` class, regardless of
+ * requested positioning.
+ */
+.md-toast-open-top-left .md-button.md-fab-top-left,
```

Comment:
it feels like all those selectors can be built by a mixin - not saying we should, just suggesting :)